packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx

```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
   }, [condition, fn, delay, timer, setTimer]);
 }
 
-type VideoPopoverProps = PropsWithChildren<{
-  open: boolean;
-  onClose: () => void;
-  onPlay?: () => void;
-  publicId: string;
-  cloudName: string;
-  delay?: number;
-  thumbnail?: string;
-  side?: HoverCardProps["side"];
-  sideOffset?: HoverCardProps["sideOffset"];
-  arrowSize?: number;
-  align?: HoverCardProps["align"];
-  alignOffset?: HoverCardProps["alignOffset"];
-}>;
-
-export const VideoPopover: FC<VideoPopoverProps> = ({
-  children,
-  onClose,
-  onPlay,
-  open,
-  delay = 5000,
-  publicId,
-  cloudName,
-  thumbnail,
-  side,
-  sideOffset,
-  arrowSize,
-  align,
-  alignOffset,
-}) => {
-  const [isOpen, setOpen] = useState<boolean>(false);
-
-  const handleClose = useCallback(() => {
-    onClose();
-    setOpen(false);
-  }, [onClose, setOpen]);
-
-  const handleDefaultOpening = useCallback(
-    () => setOpen((wasOpen) => wasOpen === false),
-    [setOpen]
-  );
-
-  useTimerWhen(open, handleDefaultOpening, delay);
-
-  const handleChange = (value: boolean) => {
-    if (value === false) {
-      handleClose();
-    } else if (value === true && open === true) {
-      setOpen(value);
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
+  ref: React.MutableRefObject<T | null>,
+  handler: () => void
+) {
+  useEffect(() => {
+    function checkWhereWasClicked(event: MouseEvent) {
+      if (
+        ref.current !== null &&
+        event.target !== null &&
+        ref.current.contains(event.target as Node) === false
+      ) {
+        handler();
+      }
     }
-  };
 
-  return (
-    <HoverCard
-      anchor={children}
-      open={isOpen}
-      onOpenChange={handleChange}
-      openDelay={delay}
-      side={side}
-      sideOffset={sideOffset}
-      arrowSize={arrowSize}
-      align={align}
-      alignOffset={alignOffset}
-    >
-      <VideoContainer
-        onClose={handleClose}
-        onPlay={onPlay}
-        cloudName={cloudName}
-        publicId={publicId}
-        thumbnail={thumbnail}
-      />
-    </HoverCard>
-  );
-};
+    document.addEventListener("mousedown", checkWhereWasClicked);
```

Comment:
Maybe we should add and remote the event listenner only when the ref is not null?

```suggestion
    if (ref.current !== null) {
      document.addEventListener("mousedown", handleClickOutside);
    }

    return () => {
      if (ref.current !== null) {
        document.removeEventListener("mousedown", handleClickOutside);
      }
    }
```

Reply:
for removing it, i'd say it shouldn't matter if it's null or not (as the ref is mutablle)
for addding yes, that makes sense :)

Reply:
Good one! 100% agree